libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -359,6 +359,10 @@ proc validateAndRelay(g: GossipSub,
           break
     toSendPeers.excl(seenPeers)
 
+    # add always direct peers
+    for topic in msg.topicIds:
+      toSendPeers.incl(g.subscribedDirectPeers.getOrDefault(topic))
```

Comment:
I'll assume this gets filtered somewhere so we don't send to peers that have already seen the message?

Reply:
It wasn't, fixed